new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -230,6 +249,78 @@ class LogFileManager {
         };
     }
 
+    /**
+     * Searches for log events based on the given search string.
+     *
+     * @param searchString The search string.
+     * @param isRegex Whether the search string is a regular expression.
+     * @param matchCase Whether the search is case-sensitive.
+     * @return An object containing the search results.
+     */
+    startQuery (searchString: string, isRegex: boolean, matchCase: boolean): void {
+        this.incrementQueryId();
+
+        // If the search string is empty, or there are no logs, return
+        if ("" === searchString) {
+            return;
+        } else if (0 === this.#numEvents) {
+            return;
+        }
+
+        // Construct search RegExp
+        const regexPattern = isRegex ?
+            searchString :
+            searchString.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+        const regexFlags = matchCase ?
+            "" :
+            "i";
+        const searchRegex = new RegExp(regexPattern, regexFlags);
+        this.#searchChunk(this.#queryId, 0, searchRegex);
+    }
+
+    /**
+     * Searches for log events in the given range.
```

Comment:
Do we want to mention that this method schedules a task for searching the next chunk after we are done querying this current chunk?

Reply:
If so, maybe we should also rename the method as `searchChunkAndScheduleNext` or something similar. What do you think?